main.tf

```diff
@@ -25,27 +25,34 @@ locals {
   default_name                   = "cloud-nat-${random_string.name_suffix.result}"
   nat_ips_length                 = length(var.nat_ips)
   default_nat_ip_allocate_option = local.nat_ips_length > 0 ? "MANUAL_ONLY" : "AUTO_ONLY"
-
   # locals for google_compute_router_nat
   nat_ip_allocate_option = var.nat_ip_allocate_option ? var.nat_ip_allocate_option : local.default_nat_ip_allocate_option
   name                   = var.name != "" ? var.name : local.default_name
+  router                 = var.router == "" ? google_compute_router.router[0].name : var.router
 }
 
-resource "google_compute_router_nat" "main" {
+resource "google_compute_router" "router" {
+  count   = var.router == "" ? 1 : 0
```

Comment:
I'm concerned that this change will introduce count issues in cases where people are passing a dynamic `router` name (ie. directly using a Terraform reference).

Could we instead have a separate `create_router` variable?

Reply:
Good call, let me change this.

Reply:
Done.